src/Nethermind/Nethermind.Evm.Test/Eip2935Tests.cs

```diff
@@ -0,0 +1,109 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using FluentAssertions;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Specs;
+using Nethermind.Specs;
+using Nethermind.Core.Test.Builders;
+using Nethermind.Int256;
+using Nethermind.Serialization.Json;
+using Nethermind.State;
+using NUnit.Framework;
+
+namespace Nethermind.Evm.Test
+{
+    public class Eip2935Tests : VirtualMachineTestsBase
+    {
+        protected override long BlockNumber => MainnetSpecProvider.BerlinBlockNumber;
+        protected override ISpecProvider SpecProvider => MainnetSpecProvider.Instance;
+
+        private IReleaseSpec _spec = new ReleaseSpec();
+        private ISpecProvider _specProvider;
+
+        [SetUp]
+        public override void Setup()
+        {
+            _specProvider = new SingleReleaseSpecProvider(_spec, 0);
+            base.Setup();
+
+            TestState.CreateAccount(IVirtualMachine.BlockhashStorage, 0);
+            Keccak codeHash = TestState.UpdateCode(new byte[1] {0});
```

Comment:
this is potentially an important change to EIP (to prevent an unusual case where an account without code has storage)